index.js

```diff
@@ -235,18 +235,28 @@ function findBestSubtitleFile(list) {
 function processFormats(formats) {
   // Formats are first filtered by video codec as well as protocol. So vp9 and av01 are filtered out as they aren't supported alongside tracks using the https protocol.
   // Formats filtered based on resolution and fps. Formats that are 1080p or higher and at the most 30fps or less than 1080p with any fps are retained.
-  const filteredFormats = formats.filter(({ acodec, format_id, fps, height, protocol, vcodec }) => filterFormatCodecs(acodec, format_id, protocol, vcodec) && filterFormatFps(fps, height))
-  .sort((prevFormat, nextFormat) => nextFormat.tbr - prevFormat.tbr);
-
-  const selected4K = filteredFormats.find((format) => (format.height === 2160 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 2160);
-  const selected1080p = filteredFormats.find((format) => (format.height === 1080 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 1080);
-  const selectedLowQuality = filteredFormats.find((format) => (format.height === 720 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 720) || filteredFormats.find((format) => (format.height <= 720 && format.acodec !== 'none')) || filteredFormats.find((format) => (format.height <= 720));
+  const filteredFormats = 
+    formats.filter(({ format_id, fps, height, vcodec }) => filterFormatCodecs(format_id, vcodec) && filterFormatFps(fps, height))
+           .sort((prevFormat, nextFormat) => nextFormat.tbr - prevFormat.tbr);
+
+  const selected4K = 
+    filteredFormats.find((format) => (format.height === 2160 && format.acodec !== 'none' && format.protocol.startsWith('http'))) ||
+    filteredFormats.find((format) => format.height === 2160 && format.protocol.startsWith('http')) || 
```

Comment:
This is basically saying we prefer videos with audio and has protocol that starts with http (not sure if http is actually a thing, probably only https) as I think the box doesn't always handle m3u8 files properly.